File: lib/py/uapi/internal/mock/MockStub.py
from typing import Dict


class MockStub:
    def __init__(self, when_function: str, when_argument: Dict[str, object],
                 then_result: Dict[str, object], allow_argument_partial_match: bool, count: int) -> None:
        self.when_function = when_function
        self.when_argument = when_argument
        self.then_result = then_result
        self.allow_argument_partial_match = allow_argument_partial_match
        self.count = count

File: lib/ts/uapi/internal/mock/MockStub.ts
export class MockStub {
    public whenFunction: string;
    public whenArgument: Record<string, any>;
    public thenResult: Record<string, any>;
    public allowArgumentPartialMatch: boolean;
    public count: number;

    constructor(
        whenFunction: string,
        whenArgument: Record<string, any>,
        thenResult: Record<string, any>,
        allowArgumentPartialMatch: boolean,
        count: number,
    ) {
        this.whenFunction = whenFunction;
        this.whenArgument = whenArgument;
        this.thenResult = thenResult;
        this.allowArgumentPartialMatch = allowArgumentPartialMatch;
        this.count = count;
    }
}
